plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -1272,6 +1279,64 @@ private void testDeletionVectorsDisabledCreateTable(String tableDefinition)
+        testDeletionVectorsEnabledWriteCheckpoint("WITH (deletion_vectors_enabled = true, checkpoint_interval = 2) AS SELECT 1 x");
+    }
+
+    private void testDeletionVectorsEnabledWriteCheckpoint(String tableDefinition)
+            throws Exception
+    {
+        try (TestTable table = newTrinoTable("deletion_vectors", tableDefinition)) {
+            assertThat((String) computeScalar("SHOW CREATE TABLE " + table.getName()))
+                    .contains("deletion_vectors_enabled = true");
+
+            String tableLocation = getTableLocation(table.getName());
+            List<DeltaLakeTransactionLogEntry> transactionLogs = getEntriesFromJson(0, tableLocation + "/_delta_log");
+            assertThat(transactionLogs.get(1).getProtocol())
+                    .isEqualTo(new ProtocolEntry(3, 7, Optional.of(Set.of("deletionVectors")), Optional.of(Set.of("deletionVectors"))));
+
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES 2, 3", 2);
+            assertUpdate("DELETE FROM " + table.getName() + " WHERE x = 2", 1);
+
+            assertThat(fileNamesIn(new URI(tableLocation).getPath(), false))
+                    .anyMatch(path -> path.matches(".*/deletion_vector_.*.bin"));
+
+            assertThat(query("SELECT * FROM " + table.getName() + " WHERE x > 1"))
+                    .matches("VALUES 3");
+
+            Location checkpointLocation = Location.of(new URI(tableLocation).getPath() + "/_delta_log").appendPath("%020d.checkpoint.parquet".formatted(2));
+            assertThat(fileNamesIn(new URI(tableLocation).getPath() + "/_delta_log", false))
+                    .anyMatch(path -> path.equals(checkpointLocation.toString()));
+
+            TrinoInputFile checkpoint = FILE_SYSTEM.newInputFile(checkpointLocation);
+            CheckpointEntryIterator checkpointEntryIterator = new CheckpointEntryIterator(
+                    checkpoint,
+                    SESSION,
+                    checkpoint.length(),
+                    new CheckpointSchemaManager(TESTING_TYPE_MANAGER),
+                    TESTING_TYPE_MANAGER,
+                    Set.of(CheckpointEntryIterator.EntryType.values()),
+                    transactionLogs.stream().map(DeltaLakeTransactionLogEntry::getMetaData).filter(Objects::nonNull).findFirst(),
+                    Optional.of(transactionLogs.get(1).getProtocol()),
+                    new FileFormatDataSourceStats(),
+                    ParquetReaderOptions.builder()
+                            .withMaxBufferSize(DataSize.ofBytes(500))
+                            .withSmallFileThreshold(DataSize.of(1, KILOBYTE))
+                            .build(),
+                    true,
+                    new DeltaLakeConfig().getDomainCompactionThreshold(),
+                    TupleDomain.all(),
+                    Optional.of(alwaysTrue()));
+            while (checkpointEntryIterator.hasNext()) {
+                checkpointEntryIterator.next();
+            }
+        }
```

Comment:
I guess future readers don't understand the intention of these code. Coud you leave a simple comment?